### ISSUE TITLE
Kernel 5.4.49 cross compile issue

### ISSUE THREAD

**Issue by Dark-Sky** [raspberrypi/linux#3693]
Compiler: arm-none-eabi-gcc (Arch Repository) 10.1.0 

Using the same .config I had for kernel 5.4.47 and it automatically adds this to my .config:
 
 CONFIG_CC_CAN_LINK=y
 
This brings up in the terminal when using "make -j6 ARCH=arm CROSS_COMPILE=arm-none-eabi-" with a ton of questions that appear it is wanting to redo my .config for 64bit x86 if I accepted the defaults. It is like it detects my host architecture and completely ignores the "make -j6 ARCH=arm CROSS_COMPILE=arm-none-eabi-" all together. Compiling native on my pi4b it still adds CONFIG_CC_CAN_LINK=y but I had no issue with it compiling. I guess it ignored it.
 
I reverted commit 5f07fd146d71753aa028b55ccd6449006e0812e6 and it cross compiled as normal:
 
 scripts/gcc-plugin.sh
@@ -1,5 +1,6 @@
#!/bin/sh
#SPDX-License-Identifier: GPL-2.0
 #exit 0 # Disable plugins
srctree=$(dirname "$0")
 
SHOW_ERROR=

**Comment by Dark-Sky** [comment]
I messed with this again today. After I had reverted 5f07fd1 it looks like it had straightened out something in my previous .confg as I went back to having 5f07fd1 back in the mix it cross compiled like it is supposed to using the new .config with "Compiler: arm-none-eabi-gcc (Arch Repository) 10.1.0".

I am in the process of starting over with make _defconfig's on all rpi 32 bit and 64 bit kernels to see if all is good before I close this as I have the pi1, pi2, pi3b, pi3b+, pi4b 4G and the pi4b 8G I like to keep everything current on.So far no problem.
